File: java/Array/LongestConsecutiveSequence.java
import java.util.Scanner;
// import java.util.Arrays;
import java.util.HashSet;

class LongestConsecutiveSequence {

    public int longestConsecutive(int[] arr) {
        // int n= arr.length;
        // int max_count=0;
        // for(int i=0;i<n;i++){
        //     int x=arr[i];
        //     int count=1;
        //     for(int j=i+1;j<n;j++){
        //         if(x+1==arr[j]){
        //             x+=1;
        //             count++;
        //             j=-1;
        //         }
        //     }
        //     max_count=Math.max(max_count,count);
        // }
        // return max_count;

        // int n= arr.length;
        // if(n==0){
        //     return 0;
        // }
        // Arrays.sort(arr);
        // int max_count=1;
        // int count=1;
        // int prev_smaller=arr[0];
        // for(int i=1;i<n;i++){
        //     if(arr[i]==prev_smaller+1){
        //         count++;
        //     }
        //     else if(arr[i]!=prev_smaller){
        //         count=1;
        //     }
        //     prev_smaller=arr[i];
        //     max_count=Math.max(max_count,count);
        // }
        // return max_count;

        int n= arr.length;
        if(n==0){
            return 0;
        }
        int max_count=1;
        HashSet<Integer> set= new HashSet<Integer>();
        for(int i=0;i<n;i++){
            set.add(arr[i]);
        }
        for(int it : set)
        {
            if(!set.contains(it-1))
            {
                int count=1;
                while(set.contains(it+1))
                {
                    it++;
                    count++;
                }
                max_count=Math.max(max_count,count);
            }
        }
        return max_count;
        }
}
class Main {
    public static void main(String[] args) {
        Scanner sc= new Scanner(System.in);
        int n= sc.nextInt();
        int[] arr= new int[n];
        for(int i=0;i<n;i++){
            arr[i]=sc.nextInt();
        }
        LongestConsecutiveSequence obj = new LongestConsecutiveSequence();
        System.out.println(obj.longestConsecutive(arr));
        sc.close();
    }    
}

File: python/Array/LongestConsecutiveSequence.py
import math
class Solution:
    def longestConsecutive(self, nums):
        # n=len(nums)
        # if n==0:
        #     return 0
        # max_count=0
        # for i in range(n):
        #     x=nums[i]
        #     count=1
        #     while x+1 in nums:
        #         x+=1
        #         count+=1
        #     max_count=max(max_count,count)
        # return max_count
        
        # n=len(nums)
        # if n==0:
        #     return 0
        # nums.sort()
        # count=1
        # max_count=1
        # prev_smallest=nums[0]
        # for i in range(1,n):
        #     if nums[i]==prev_smallest+1:
        #         count+=1
        #     else:
        #         max_count=max(max_count,count)
        #         count=1
        #     prev_smallest=nums[i]
        # return max(max_count,count)
        
        n=len(nums)
        if n==0:
            return 0
        
        max_count=1
        sett=set(nums)
        for i in range(n):
            if nums[i]-1 not in sett:
                x=nums[i]
                count=1
                while x+1 in sett:
                    x+=1
                    count+=1
                max_count=max(max_count,count)
        return max_count
        

nums=list(map(int,input().split()))
obj=Solution()
result=obj.longestConsecutive(nums)
print(result)
